### PR TITLE
Simplify installation of Rust in osbuilder

### PR DESCRIPTION
**Pull request by dgibson**
We attempt to install rust in more circumstances than we need to, and the handling is overly complex.

### CODE REVIEW

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (jodh-intel) [inline tools/osbuilder/rootfs-builder/rootfs.sh:47]: Nit: How about setting `AGENT_VERSION` to simplify this and avoid duplicated defaults?

```suggestion
AGENT_VERSION=${AGENT_VERSION:-main}

get_kata_rust_version ||
    die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION}'."
get_kata_musl_version ||
    die "Could not detect the required musl version for AGENT_VERSION='${AGENT_VERSION}'."
```

Same comment for the calls in `test_images.sh`.

**Review by jodh-intel — APPROVED** [review]
Thanks @dgibson.

**Comment by jcvenegas** [comment]
@dgibson  this breaks the docker build, 
```
INFO: Configure chrony file /rootfs/etc/chrony.conf
/kata-containers/tools/osbuilder/rootfs-builder/rootfs.sh: line 557: rustup: command not found
ERROR: Rust target linux-musl not installed
make: *** [Makefile:93: /home/runner/work/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/rootfs-initrd/builddir/initrd-image/.alpine_rootfs.done] Error 1
make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:30: rootfs-initrd-tarball-build] Error 2
make[1]: Leaving directory '/home/runner/work/kata-containers/kata-containers'
make: *** [tools/packaging/kata-deploy/local-build/Makefile:52: rootfs-initrd-tarball] Error 2
```
See details on:
https://github.com/kata-containers/kata-containers/runs/4208158662?check_suite_focus=true

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (wainersm) [reply tools/osbuilder/rootfs-builder/rootfs.sh:47]: And in the lines 47 and 49: `s/detect/get/`

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (wainersm) [inline tools/osbuilder/rootfs-builder/rootfs.sh:566]: This change will break some CI jobs.

I did grep the tests repository, it turned out that `tests/.ci/jenkins_job_build.sh` and `tests/tracing/test-agent-shutdown.sh` are the only scripts calling the `install_rust.sh` script. The former is used by most of jobs on Jenkins ( that's why some jobs ran fine with that change) but others won't use it; examples are  `tests/.ci/openshift-ci/build_install.sh` (used on OpenShift CI) and `tests/.ci/vfio_jenkins_job_build.sh` (entry point of the VFIO job). Also if I run a CI job locally with `.ci/setup.sh && .ci/run.sh` it will fail too (I guess).

I don't like that Kata scripts install stuffs on developer's system so I find this change welcomed. We just need to find the best place to call the `install_rust.sh` script to disturb the fewest jobs (use cases) possible.

Comment (dgibson) [reply tools/osbuilder/rootfs-builder/rootfs.sh:566]: Right, I noticed the problem, I just haven't had a chance to debug yet (hence `wip`).  I'm thinking I'll probably take a different approach to this whole thing.

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (dgibson) [reply tools/osbuilder/rootfs-builder/rootfs.sh:47]: > Nit: How about setting `AGENT_VERSION` to simplify this and avoid duplicated defaults?

There are other places which use `AGENT_VERSION` and expect the default value to be empty.  Disentangling that isn't really in scope for this patchset.

Comment (dgibson) [reply tools/osbuilder/rootfs-builder/rootfs.sh:47]: > And in the lines 47 and 49: `s/detect/get/`

I've made this change.

**Comment by jodh-intel** [comment]
Any update on this @dgibson? Should this be closed?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by fidencio** [comment]
As this PR has been waiting for an update since November 2021, I'm closing it for now.

@dgibson, please, if you find some free cycles to work on this PR, please, just re-open it and let us know.
Thanks for the contribution and looking forward to seeing an up-to-date version of this PR.

**Pull request closed without merging**
